### PR TITLE
Add support for metadata.

### PR DESCRIPTION
**Pull request by functicons**

### CODE REVIEW

File: custom_image_utils/shell_script_generator.py [hunk 1]
```diff
@@ -162,6 +162,10 @@ def _init_args(self, args):
     self.args[
       "storage_location_flag"] = "--storage-location={storage_location}".format(
         **self.args) if self.args["storage_location"] else ""
+    if self.args["metadata"]:
+      self.args["metadata_flag"] = "--metadata=shutdown-timer-in-sec={shutdown_timer_in_sec},custom-sources-path={custom_sources_path},{metadata}".format(**self.args)
+    else:
+      self.args["metadata_flag"] = "--metadata=shutdown-timer-in-sec={shutdown_timer_in_sec},custom-sources-path={custom_sources_path}".format(**self.args)
```
Comment (medb) [inline custom_image_utils/shell_script_generator.py:168]: De-duplicate?
```python
metadata_flag = "--metadata=shutdown-timer-in-sec={shutdown_timer_in_sec},custom-sources-path={custom_sources_path}"
if self.args["metadata"]:
  metadata_flag += ",{metadata}"
self.args["metadata_flag"] = metadata_flag.format(**self.args)
```

File: README.md [hunk 2]
```diff
@@ -134,6 +141,7 @@ python generate_custom_image.py \
     --image-name custom-image-1-4-5 \
     --dataproc-version 1.4.5-debian9 \
     --customization-script ~/custom-script.sh \
+    --metadata 'key1=value1,key2=value2'
```
[the remark this answers is not in the document]
Comment (functicons) [reply README.md]: Done

File: custom_image_utils/shell_script_generator.py (hunk 1 above)
Comment (functicons) [reply custom_image_utils/shell_script_generator.py]: Done.

**Pull request merged**